Desktop/components/JASP/Widgets/FileMenu/Database.qml

```diff
@@ -0,0 +1,379 @@
+                    nextEl:					connectButton
+                    text:					fileMenuModel.database.password
+                    onTextChanged:			fileMenuModel.database.password = text;
+
+                    textInput.echoMode:		TextInput.Password
+
+                    x:						dbHostnameLabel.width
+                    width:					dbHostnameInput.width
+                }
+            }
+
+            RoundedButton
+            {
+                id:					connectButton
+                text:				qsTr("Connect to database")
+                onClicked:			fileMenuModel.database.connect();
+                KeyNavigation.tab:	dbQuery.textInput
+            }
+
+
+            RowLayout
+            {
+                width:						parent.width
+
+                enabled:					fileMenuModel.database.connected
+
+                Text
+                {
+                    id:						dbQueryLabel
+                    text:					qsTr("Query")
+                    width:					implicitWidth + jaspTheme.generalAnchorMargin
+                    //anchors.verticalCenter: parent.verticalCenter
+                }
+
+                PrefsTextInput
+                {
+                    id:						dbQuery
+                    nextEl:					runQuery
+                    text:					fileMenuModel.database.query
+                    onTextChanged:			fileMenuModel.database.query = text
+                    LQ.Layout.fillWidth:	true
+                }
+
+                RoundedButton
+                {
+                    id:						runQuery
+                    text:					qsTr("Execute")
+                    onClicked:				fileMenuModel.database.runQuery();
+                    KeyNavigation.tab:		loadResults
+                }
+            }
+
+        }
+
+        PrefsGroupRect
+        {
+            id:			previewGroup
+            title:		qsTr("Preview data")
+            enabled:	fileMenuModel.database.connected && fileMenuModel.database.resultsOK
+
+            Item
+            {
+                height:			childrenRect.height
+                anchors
+                {
+                    left:		parent.left
+                    right:		parent.right
+                    margins:	jaspTheme.generalAnchorMargin
+                }
+
+                RoundedButton
+                {
+                    id:						loadResults
+                    text:					qsTr("Load into JASP")
+                    onClicked:				fileMenuModel.database.importResults();
+                    KeyNavigation.tab:		intervalSpinner
+                    anchors.left:			parent.left
+                }
+
+                SpinBox
+                {
+                    id:					intervalSpinner
+                    value:				fileMenuModel.database.interval
+                    onValueChanged:		if(value != "") fileMenuModel.database.interval = value
+                    from:				0
+                    to:					24 * 60 * 7 // do we want to allow for an interval bigger than a week?
```

Comment:
Why to put a max limitation? It's not necessary, you never know how JASP will be used in te future.

Reply:
I guess, I will remove it.